Source/Core/DiscIO/Volume.h

```diff
@@ -34,8 +34,8 @@ struct Partition final
   constexpr bool operator==(const Partition& other) const { return offset == other.offset; }
   constexpr bool operator<(const Partition& other) const { return offset < other.offset; }
   constexpr bool operator>(const Partition& other) const { return other < *this; }
-  constexpr bool operator<=(const Partition& other) const { return !(*this < other); }
-  constexpr bool operator>=(const Partition& other) const { return !(*this > other); }
+  constexpr bool operator<=(const Partition& other) const { return *this >= other; }
+  constexpr bool operator>=(const Partition& other) const { return *this <= other; }
```

Comment:
It looks like these `operator<=` and `operator>=` definitions were incorrect from the start (not your fault).

Consider just replacing **all** the operators here with just:
```cpp
  constexpr bool operator==(Partition other) const { return offset == other.offset; }
  constexpr auto operator<=>(Partition other) const { return offset <=> other.offset; }
```